### PR TITLE
Manual: minimal documentation for compiler plugins

### PR DESCRIPTION
**Pull request by Octachron**
This PR adds a new chapter to the tool part of the manual in order to provide some minimal documentation on the new plugins infrastrucutre (beyond the laconic mention in the man pages).

Currently, the proposed documentation only contains a brief description of the available hooks in the compiler and a basic "hello world" plugin. This is still a very basic starting point, so I am open to any improvement propositions.

### CODE REVIEW

File: manual/manual/cmds/plugins.etex [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    @@ Location.mknoloc @@Longident.Lident "print_endline" in
+    let hello = Exp.constant @@ Const.string @@ "Hello from: " ^ name in
+    Str.eval @@ Exp.apply print_endline [Asttypes.Nolabel, hello]
+\end{verbatim}%
+Then, we can construct an ast mapper that adds this fragment to the parsed
+ast tree.
+\begin{verbatim}
+let add_hello name (mapper:Ast_mapper.mapper) structure =
+  let default = Ast_mapper.default_mapper in
+  (print_ensline name) :: (default.structure default structure)
```
Comment (hcarty) [inline manual/manual/cmds/plugins.etex]: Typo? `s/ensline/endline/`

File: manual/manual/cmds/plugins.etex [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+let ast_mapper name =
+  { Ast_mapper.default_mapper with structure = add_hello name }
+\end{verbatim}%
+%
+Once this AST mapped constructed, we need to convert it to a hook and adds this
```
Comment (hcarty) [inline manual/manual/cmds/plugins.etex]: Typo? `s/mapped constructed/mapper is constructed/`

File: manual/manual/cmds/plugins.etex [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+hooks to the "Pparse.ImplementationsHooks".
```
Comment (hcarty) [inline manual/manual/cmds/plugins.etex]: Typo? `s/hooks/hook/`

Comment (Octachron) [reply manual/manual/cmds/plugins.etex]: Fixed, thanks.

**Comment by Drup** [comment]
Maybe you should list the various hooks there too ?

**Comment by Octachron** [comment]
@Drup, good idea, thanks for the suggestion: 

I have added links or references to the `Pparse`, `Typemod`  and `Simplif` modules and pulled them into the `compiler-libs` documentation since they provide plugins hooks.

I have also pulled the `Clflags` module to mention the possibility to defnine plugin specific arguments.

**Comment by gasche** [comment]
Merged, thanks! We would really a better separation between the all-users-facing standard library and the advanced-user-facing parts in the manual.

**Pull request merged**